protostar/testing/test_environment_exceptions.py

```diff
@@ -281,3 +281,28 @@ def __reduce__(self):
             ),
             self.__getstate__(),
         )
+
+
+class ExpectedCallException(ReportedException):
+    def __init__(
+        self, contract_address: int, fn_name: str, calldata: list[int]
+    ) -> None:
+        self._contract_address = contract_address
+        self._calldata = calldata
+        self._fn_name = fn_name
+        super().__init__()
+
+    def __str__(self) -> str:
+        return (
+            f"expected call to function {self._fn_name} "
+            f"from the contract of address {hex(self._contract_address)} "
+            f"with calldata {self._calldata} not fulfilled.\n"
+            f"Please, keep in mind that you can use expect_call only on deployed contracts"
```

Comment:
The purpose of this cheatcode is actually using in on not-deployed contracts in order to allow testing interaction with contracts outside the user's control, e.g. oracle contracts.

I think this is not a right place for this message, because a test will break on calling a non-deployed contract.
Please create a new issue for that and a warning in the docs.

Reply:
I can also just remove the last sentence here

Reply:
[don't show calling expect_call after the call in the docs](https://github.com/software-mansion/protostar/pull/1060/commits/974083f1225c6aa8ce800b2a53b94f927ba47dc2)

Reply:
Hmm. Actually, `expect_call` can be used with `mock_call`. It would be worth to create an issue to limit the scope in which the call is expected (if it's feasible).

```
stop_expecting_call = expect_call(...)
   ...
stop_expecting_call()
```

Reply:
#1126